plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/stream/StreamTransferConsumer.java

```diff
@@ -1061,23 +1074,37 @@ public void writeBinaryData(@NotNull DBDContentStorage cs) throws IOException {
                                 IOUtils.copyStream(stream, buffer);
 
                                 final byte[] bytes = buffer.toByteArray();
-                                final String binaryString = dataSource.getSQLDialect().getNativeBinaryFormatter().toString(bytes, 0, bytes.length);
+                                String binaryString = dataSource.getSQLDialect().getNativeBinaryFormatter().toString(bytes, 0, bytes.length);
+                                if (DBConstants.TYPE_NAME_JSON.equalsIgnoreCase(typeOfOutput)) {
+                                    binaryString = CommonUtils.getQuotedString(binaryString);
+                                }
                                 writer.write(binaryString);
                                 break;
                             }
                         }
                         default: {
                             // Binary stream
                             try (Reader reader = new InputStreamReader(stream, cs.getCharset())) {
+                                String string = IOUtils.readToString(reader);
```

Comment:
This is a big no. It might be a really big blob (like several GBs in size) that we would otherwise just pipe to the output stream (writer).